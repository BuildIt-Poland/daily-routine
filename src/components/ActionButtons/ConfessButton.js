import React from 'react';
import styled from 'styled-components';

import { role } from '../../types';
import { CONFESS_ROLE_ACTION } from '../../constants/roles';
import { colorRed, colorDarkRed } from '../../styles/designTokens';
import SaltGrinderIcon from './SaltGrinderIcon';
import ButtonBase from './ButtonBase';

const StyledButton = styled(ButtonBase)`
  background-color: ${colorRed};
  border-bottom-color: ${colorDarkRed};
`;

function ConfessButton({ role, ...props }) {
  return (
    <StyledButton to={`/${role}/${CONFESS_ROLE_ACTION}`} icon={SaltGrinderIcon} {...props}>
      Confess my mistake
    </StyledButton>
  );
}

ConfessButton.propTypes = {
  role: role.isRequired
};

export default ConfessButton;
