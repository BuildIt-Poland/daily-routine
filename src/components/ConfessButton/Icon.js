import React from 'react';
import styled from 'styled-components';

import { SaltGrinder } from '../Icons';

const Wrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: -1rem;
`;

function Icon() {
  return (
    <Wrapper>
      <SaltGrinder />
    </Wrapper>
  );
}

export default Icon;
