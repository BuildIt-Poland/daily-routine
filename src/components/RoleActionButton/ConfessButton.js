import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { colorRed, colorDarkRed } from '../../styles/designTokens';
import Button from '../Button';
import SaltGrinderIcon from './SaltGrinderIcon';

const StyledButton = styled(Button)`
  background-color: ${colorRed};
  border-bottom-color: ${colorDarkRed};
`;

function ConfessButton({ to }) {
  return (
    <StyledButton as={Link} to={to}>
      Confess my mistake
      <SaltGrinderIcon />
    </StyledButton>
  );
}

ConfessButton.propTypes = {
  to: PropTypes.string.isRequired
};

export default ConfessButton;
