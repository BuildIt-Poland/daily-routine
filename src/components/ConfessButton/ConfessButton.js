import React from 'react';
import styled from 'styled-components';

import { colorRed, darkRed } from '../../styles/designTokens';
import WideButton from '../WideButton';
import Icon from './Icon';

const StyledButton = styled(WideButton)`
  background-color: ${colorRed};
  border-bottom-color: ${darkRed};
`;

function ConfessButton() {
  return <StyledButton icon={<Icon />}>Confess my mistake</StyledButton>;
}

export default ConfessButton;
