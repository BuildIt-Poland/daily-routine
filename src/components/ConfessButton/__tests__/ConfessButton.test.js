import React from 'react';
import { create } from 'react-test-renderer';

import ConfessButton from '../ConfessButton';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

jest.mock('../../Icons', () => ({
  SaltGrinder: 'SaltGrinder'
}));

describe('COMPONENT - ConfessButton', () => {
  it('renders correctly', () => {
    const component = create(<ConfessButton />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
