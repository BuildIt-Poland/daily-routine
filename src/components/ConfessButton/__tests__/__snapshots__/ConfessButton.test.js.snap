// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - ConfessButton renders correctly 1`] = `
.c1 {
  position: relative;
  background: #21c230;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #138a1e;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;
}

.c1:not(:disabled) {
  cursor: pointer;
}

.c2 {
  position: absolute;
  right: 0.4rem;
  bottom: -0.9rem;
}

.c0 {
  background: #eb4640;
  border-bottom-color: #a32f2a;
}

<Link
  className="c0 c1"
  to="/"
>
  Confess my mistake
  <div
    className="c2"
  >
    <SaltGrinder />
  </div>
</Link>
`;
